src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+  static removeAnimal() {
+    Animal.alive = Animal.alive.filter((animalItem) => animalItem.health > 0);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden === false) {
+      animal.health -= 50;
+
+      if (animal.health === 0) {
+        Animal.removeAnimal();
```

Comment:
Invoking removeAnimal directly inside the bite method is not efficient because it will be called every time a bite occurs, even if the animal's health is not 0. It would be better to call removeAnimal after all bite actions are completed, possibly in a game loop or similar structure.